### DISCUSSION TITLE
Business rules for Affected properties

### DISCUSSION THREAD

**Discussion by Chicagoan2016** [MarimerLLC/csla#3394] [Questions]
In one of our objects, we have Properties that are never set by UI. They are calculated by business rules using context.AddOutValue. 
We tried to write business rules for those 'calculated' properties but the business rule don't run. I am assuming context.AddOutValue doesn't call SetProperty.
Any ideas please.

Regards

**Comment by rockfordlhotka** [comment]
Rules are triggered either by a change to the primary property, or when `CheckRules` is called to run all rules.

If changing one property will affect other property values, those properties need to be in the affected properties list for the rule, and that rule is triggered as described above.

Of course @jonnybee is the ultimate authority on all of this 🙂

**Reply by Chicagoan2016** [reply]
In my case, since the Affected properties are only set by context.AddOutValue, the rules for those 'affected' properties are not called.
